test/e2e/resource_fairness/resource_fairness_test.go

```diff
@@ -191,14 +248,203 @@ var _ = Describe("FairScheduling:", func() {
+		RespPod, err = kClient.GetPod(sleepPod4Conf.Name, ns)
+		Ω(err).NotTo(HaveOccurred())
+		Ω(RespPod.Spec.NodeName).To(gomega.Equal(workerID2), "Pod should place on "+workerID2)
+	})
+
+	// Validates the order of node allocation for requested pod resources, considering fairNodeScheduling with resource weights.
+	// Step 1: Set the Resource weights to {"vcore":  5.0,	"memory": 1.0}
+	// Step 2: Deploy 2 apps, increase resource utilization on both worker1 and worker2. It's expected that worker2 will experience higher resource utilization.
+	// Step 3: Deploy 1 apps is expected to allocate it to worker1, Leading to higher resource utilization on worker1.
+	// Step 4: Deploy 1 apps is expected to allocate it to worker2.
+	It("Verify_node_sorting_fairness_policy_with_resource_weight", func() {
+		By("Setting custom YuniKorn configuration")
+		annotation = "ann-" + common.RandSeq(10)
+		queuePath = "root.sandbox"
+		yunikorn.UpdateCustomConfigMapWrapper(oldConfigMap, "fair", annotation, func(sc *configs.SchedulerConfig) error {
+			// remove placement rules so we can control queue
+			sc.Partitions[0].PlacementRules = nil
+			sc.Partitions[0].NodeSortPolicy = configs.NodeSortingPolicy{
+				Type: "fair",
+				ResourceWeights: map[string]float64{
+					"vcore":  5.0,
+					"memory": 1.0,
+				},
+			}
+			if err = common.AddQueue(sc, "default", "root", configs.QueueConfig{
+				Name:   "sandbox",
+				Parent: false,
+			}); err != nil {
+				return err
+			}
+			return nil
+		})
+
+		// Assign pod1 to worker1 and pod2 to worker2.
+		sleepPod1Conf := k8s.SleepPodConfig{
+			Name: "pod1", NS: ns, AppID: "app1",
+			RequiredNode: workerID1,
+			Labels:       map[string]string{constants.LabelQueueName: queuePath}}
+		sleepPod2Conf := k8s.SleepPodConfig{
+			Name: "pod2", NS: ns, AppID: "app2",
+			RequiredNode: workerID2,
+			Labels:       map[string]string{constants.LabelQueueName: queuePath}}
+
+		// Verify if pod3 and pod4 are assigned based on node sorting fairness.
+		sleepPod3Conf := k8s.SleepPodConfig{
+			Name: "pod3", NS: ns, AppID: "app3",
+			Labels: map[string]string{constants.LabelQueueName: queuePath}}
+		sleepPod4Conf := k8s.SleepPodConfig{
+			Name: "pod4", NS: ns, AppID: "app4",
+			Labels: map[string]string{constants.LabelQueueName: queuePath}}
+
+		sleepPodConfs := []k8s.SleepPodConfig{}
+		sleepPodConfs = append(sleepPodConfs, sleepPod1Conf)
+		sleepPodConfs = append(sleepPodConfs, sleepPod2Conf)
+		sleepPodConfs = append(sleepPodConfs, sleepPod3Conf)
+		sleepPodConfs = append(sleepPodConfs, sleepPod4Conf)
+
+		// The node where the pod is expected to be deployed affects its resource utilization.
+		expectedNode := []string{
+			workerID1,
+			workerID2,
+			workerID1,
+			workerID2,
+		}
+		increaseUtiliz := []map[string]float64{
+			{"vcore": 0.05, "memory": 0.10},
+			{"vcore": 0.10, "memory": 0.05},
+			{"vcore": 0.15, "memory": 0.15},
+			{"vcore": 0.15, "memory": 0.15},
```

Comment:
Please extract constants, not just here, but everywhere in this file. These literals are repeated way too often.